File: season_1/programming_lab/TDE_01/exercise_04.py
try:
    numero = int(input("Digite um número de 3 dígitos: "))

    if 100 <= abs(numero) <= 999:
        centenas = abs(numero) // 100

        if centenas % 2 == 0:
            print("O algarismo das centenas é par.")
        else:
            print("O algarismo das centenas é ímpar.")
    else:
        print("Erro: o número não tem 3 dígitos.")

except ValueError:
    print("Erro: você não digitou um número inteiro válido.")

File: season_1/programming_lab/TDE_01/exercise_04.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    int numero;

    printf("Digite um número de 3 dígitos: ");

    if (scanf("%d", &numero) != 1) {
        printf("Erro: voce nao digitou um numero inteiro valido.\n");
        return 1;
    }

    if (abs(numero) >= 100 && abs(numero) <= 999) {
        int centenas = abs(numero) / 100;

        if (centenas % 2 == 0) {
            printf("O algarismo das centenas e par.\n");
        } else {
            printf("O algarismo das centenas e impar.\n");
        }
    } else {
        printf("Erro: o numero nao tem 3 digitos.\n");
    }

    return 0;
}